engine/table/src/main/java/io/deephaven/engine/util/ColumnFormatting.java

```diff
@@ -0,0 +1,125 @@
+     * @return The name of the row style formatting column
+     */
+    static String getRowStyleFormatColumn() {
+        return Constants.ROW_STYLE_FORMAT_COLUMN;
+    }
+
+    /**
+     * Returns true if this column name is a hidden row style formatting column.
+     *
+     * @param columnName the column name to check
+     * @return true if the columnName is a row style formatting column; false otherwise
+     */
+    static boolean isRowStyleFormatColumn(@NotNull final String columnName) {
+        return columnName.equals(Constants.ROW_STYLE_FORMAT_COLUMN);
+    }
+
+    /**
+     * @param baseColumn the column name
+     * @return The name of the style formatting column for the specified base column.
+     */
+    static String getStyleFormatColumn(String baseColumn) {
+        return baseColumn + Constants.TABLE_STYLE_FORMAT_SUFFIX;
+    }
+
+    /**
+     * Returns true if this column name is a hidden style formatting column.
+     *
+     * @param columnName the column name to check
+     * @return true if the columnName is a style formatting column; false otherwise
+     */
+    static boolean isStyleFormatColumn(@NotNull final String columnName) {
+        return columnName.endsWith(Constants.TABLE_STYLE_FORMAT_SUFFIX);
+    }
+
+    /**
+     * @param baseColumn the column name
+     * @return The name of the number formatting column for the specified base column.
+     */
+    static String getNumberFormatColumn(String baseColumn) {
+        return baseColumn + Constants.TABLE_NUMBER_FORMAT_SUFFIX;
+    }
+
+    /**
+     * Returns true if this column name is a hidden number formatting column.
+     *
+     * @param columnName the column name to check
+     * @return true if the columnName is a number formatting column; false otherwise
+     */
+    static boolean isNumberFormatColumn(@NotNull final String columnName) {
+        return columnName.endsWith(Constants.TABLE_NUMBER_FORMAT_SUFFIX);
+    }
+
+    /**
+     * @param baseColumn the column name
+     * @return The name of the date formatting column for the specified base column.
+     */
+    static String getDateFormatColumn(String baseColumn) {
+        return baseColumn + Constants.TABLE_DATE_FORMAT_SUFFIX;
+    }
+
+    /**
+     * Returns true if this column name is a hidden date formatting column.
+     *
+     * @param columnName the column name to check
+     * @return true if the columnName is a date formatting column; false otherwise
+     */
+    static boolean isDateFormatColumn(@NotNull final String columnName) {
+        return columnName.endsWith(Constants.TABLE_DATE_FORMAT_SUFFIX);
+    }
+
+    /**
+     * Returns the base column name from a formatting column name.
+     *
+     * @param columnName the column name
+     * @return the base column name formatted by {@code columnName}
+     */
+    static String getFormatBaseColumn(@NotNull final String columnName) {
+
+        if (columnName.startsWith(Constants.ROW_FORMAT_NAME)) {
+            return "*";
```

Comment:
So you have this "__ROW" constant; but then you return "*" and have hardcoded star elsewhere.

Reply:
Well, less me than "the code base". This is one of those scenarios where I tried to clean things up a bit without going "all in." In this particular case, I just moved the code to a single place from `SelectColumnFactory`.

To the best of my understanding, we use the "__ROW" prefix internally, since "*" isn't a valid column name, but use the "*" name more externally and expect users to input that.